libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventMapper.kt

```diff
@@ -75,10 +74,7 @@ internal object HistoryEventMapper {
         return if (routeResponse.isNullOrEmpty() || routeResponse == "{}") {
             null
         } else {
-            val gson = GsonBuilder()
-            gson.registerTypeAdapterFactory(DirectionsAdapterFactory.create())
-            gson.registerTypeAdapter(Point::class.java, PointAsCoordinatesTypeAdapter())
-            return gson.create().fromJson(routeResponse, DirectionsRoute::class.java)
```

Comment:
I just introduced this issue here https://github.com/mapbox/mapbox-navigation-android/pull/4601

Reply:
This created more discussion though. Adding it here

nav-native gives json that needs to be parsed by the SDK. it will crash the regular `DirectionsResponse.fromJson(jsonString)` because nav-native [includes routeOptions](https://github.com/mapbox/mapbox-java/blob/692c855771e6eccf25c5213e1c81dc2301d0e2df/services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java#L183-L188)

In order to parse the DirectionsRoute it works if we use the `DirectionsRoute.fromJson(routeResponse, accessToken)` version. But then that means we need to pass the `accessToken` to the `MapboxHistoryReader`

Reply:
@kmadsen Nav Native does not return route options in regular onboard route requets (fixed recently https://github.com/mapbox/mapbox-navigation-native/pull/3364), we append them in https://github.com/mapbox/mapbox-navigation-android/blob/88c20e923549c5178ad0f85bf6cc2471b35b9896/libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouter.kt#L148-L150

If route options are still visible in history events, then that's a separate bug that should be fixed.

Reply:
This test: [MapboxHistoryTest.verify_history_files_are_recorded_and_readable](https://github.com/mapbox/mapbox-navigation-android/blob/a99b4a61acdac3360bce8c30793e6d0e20b9d603/instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/MapboxHistoryTest.kt#L91) records a ride and then reads it. That test includes a route with `routeOptions`. So it seems, `MapboxNavigation` would need to strip them before setting a route.

It is still unclear to me that we do want to strip the route options as it has the request information

Reply:
We do not want to strip the options. It's just that the access token is not serialized anymore. This means that if you request a route and serialize it, then deserialize it, the `route.routeOptions` won't have the token. You would not be able to do:
```
MapboxNavigation#requestRoute(DirectionsRoute.fromJson(routeJson).routeOptions)
```
because the token is non-nullable and not serialized, so wouldn't be available. That's why the requirement to provide the token to be able to correctly deserialize the object.

Reply:
Capturing some of the feedback on the interface here https://github.com/mapbox/mapbox-navigation-android/issues/4662

We want to remove the access token from the MapboxHistoryReader. Considering some options to do so